modules/frontend/combiner/trace_by_id.go

```diff
@@ -83,6 +84,12 @@ func (c *traceByIDCombiner) AddResponse(r PipelineResponse) error {
 
 	// Consume the trace
 	_, err = c.c.Consume(resp.Trace)
+
+	if errors.Is(err, trace.ErrTraceTooLarge) {
```

Comment:
Maybe I'm testing it wrongly but I get a 500 if the frontend combiner returns an ErrTraceTooLarge here.

The error is handled here: https://github.com/grafana/tempo/blob/280128f7b740919a7e1c3ecc0ef9fd6b9843ca41/modules/frontend/handler.go#L85

Reply:
ServeHttp -> RoundTrip -> TraceByIDHandler 

The response from the TraceByIdHandler is (nil, ErrTraceTooLarge)

Reply:
oh yeah, we need to return nil in this if block. the issue is that it's falling through and still returning the error on line 93